_official-jokes/app/routes/jokes/new.tsx

```diff
@@ -32,30 +33,18 @@ function validateJokeName(name: string) {
   }
 }
 
-type ActionData = {
-  formError?: string;
-  fieldErrors?: { name: string | undefined; content: string | undefined };
-  fields?: {
-    name: string;
-    content: string;
-  };
-};
-
-/**
- * This helper function gives us typechecking for our ActionData return
- * statements, while still returning the accurate HTTP status, 400 Bad Request,
- * to the client.
- */
-const badRequest = (data: ActionData) => json(data, { status: 400 });
```

Comment:
i think we still need this 😬

Reply:
edit: never mind, new file 🤪

Reply:
Needed to extract it into a separate file if I wanted a generic version (which is needed for type inference), otherwise TS got mad 😢